src/ol_infrastructure/applications/forum/__main__.py

```diff
@@ -0,0 +1,278 @@
+                        "write_files": [
+                            {
+                                "path": "/etc/consul.d/02-autojoin.json",
+                                "content": json.dumps(
+                                    {
+                                        "retry_join": [
+                                            "provider=aws tag_key=consul_env "
+                                            f"tag_value={consul_dc}"
+                                        ],
+                                        "datacenter": consul_dc,
+                                    }
+                                ),
+                                "owner": "consul:consul",
+                            },
+                            {
+                                "path": "/etc/default/vector",
+                                "content": textwrap.dedent(
+                                    f"""\
+                            ENVIRONMENT={consul_dc}
+                            VECTOR_CONFIG_DIR=/etc/vector/
+                            AWS_REGION={aws_config.region}
+                            GRAFANA_CLOUD_API_KEY={grafana_credentials['api_key']}
+                            GRAFANA_CLOUD_PROMETHEUS_API_USER={grafana_credentials['prometheus_user_id']}
+                            GRAFANA_CLOUD_LOKI_API_USER={grafana_credentials['loki_user_id']}
+                            """
+                                ),
+                                "owner": "root:root",
+                            },
+                        ]
+                    },
+                    sort_keys=True,
+                )
+            ).encode("utf8")
+        ).decode("utf8")
+    ),
+)
+auto_scale_config = forum_config.get_object("auto_scale") or {
+    "desired": 1,
+    "min": 1,
+    "max": 2,
+}
+asg_config = OLAutoScaleGroupConfig(
+    asg_name=f"forum-server-{env_name}",
+    aws_config=aws_config,
+    desired_size=auto_scale_config["desired"],
+    min_size=auto_scale_config["min"],
+    max_size=auto_scale_config["max"],
+    vpc_zone_identifiers=target_vpc["subnet_ids"],
+    tags=aws_config.merged_tags({"Name": forum_server_tag}),
+)
+lb_config = OLLoadBalancerConfig(
+    subnets=target_vpc["subnet_ids"],
+    security_groups=[forum_server_security_group],
+    tags=aws_config.merged_tags({"Name": forum_server_tag}),
+)
+tg_config = OLTargetGroupConfig(
+    vpc_id=vpc_id,
+    health_check_interval=60,
+    health_check_matcher="200-399",
+    health_check_path="/",
+    tags=aws_config.merged_tags({"Name": forum_server_tag}),
+)
+as_setup = OLAutoScaling(
+    asg_config=asg_config,
+    lt_config=lt_config,
+    tg_config=tg_config,
+    lb_config=lb_config,
+)
+five_minutes = 60 * 5
+route53.Record(
+    "forum-server-dns-record",
+    name=forum_config.require("web_host_domain"),
+    type="CNAME",
+    ttl=five_minutes,
+    records=[as_setup.load_balancer.dns_name],
+    zone_id=mitodl_zone_id,
+)
+route53.Record(
```

Comment:
You don't need this unless forum requires a second domain for the authorization service. This was something specific for Pomerium when it was put in front of airbyte.